server.js

```diff
@@ -0,0 +1,130 @@
+        "year_created": 2023
+    },
+    {
+        "repo_name": "Weather App",
+        "details": "A weather forecasting application using React and Node.js.",
+        "repo_author": "Liam Patterson",
+        "commits": 75,
+        "year_created": 2022
+    },
+    {
+        "repo_name": "E-commerce Backend",
+        "details": "Backend code for an e-commerce platform built with Django.",
+        "repo_author": "Emily Martinez",
+        "commits": 340,
+        "year_created": 2021
+    },
+    {
+        "repo_name": "Portfolio Website",
+        "details": "Personal portfolio website showcasing projects and blogs.",
+        "repo_author": "Shurahbeel Peerzada",
+        "commits": 90,
+        "year_created": 2024
+    },
+    {
+        "repo_name": "Data Visualization Tool",
+        "details": "A tool for visualizing complex data sets using D3.js.",
+        "repo_author": "Nina West",
+        "commits": 125,
+        "year_created": 2023
+    }
+]
+
+app.listen(PORT, () => console.log(`Running on port number: ${PORT}`));
+
+app.post('/login', (req, res) => {
+    const { username, password } = req.body;
+    if (username === USERNAME && password === PASSWORD) {
+        res.cookie('auth', 'loggedin', { httpOnly: true });
+        res.status(201).send("User Successfully Logged in");
+    } else {
+        res.status(404).send("No user found with these credentials");
+    }
+});
+
+app.get('/getallrepos', (req, res) => {
+    if (req.cookies.auth === "loggedin") {
+        res.status(200).send(repo_data); 
+    } else {
+        res.status(401).send("Unauthorized Access. Please Login");
+    }
+});
+
+app.get('/getrepodetail', (req, res) => {
+    if (req.cookies.auth === 'loggedin') {
+        let name = req.query.name;
+        if (name) {
+            if (specialwords(name)) {
+                res.status(400).send("Please avoid using special characters");
+            } else {
+                let specificRepo = repo_data.find(repo => repo.repo_name === name);
+                if (specificRepo === undefined) {
+                    res.status(404).send("No repo matching the provided name exists");
+                } else {
+                    res.status(200).send(specificRepo);
+                }
+            }
+        } else {
+            res.status(400).send("Please provide a repo name");
+        }
+    } else {
+        res.status(401).send("Unauthorized Access. Please Login");
+    }
+});
+
+app.post('/create-repo', (req, res) => {
+    if (req.cookies.auth === 'loggedin') {
+        const data = req.body;
+
+        if (specialwords(data.repo_name) || specialwords(data.repo_author) || specialwords(data.details)) {
+            res.status(400).send("Please avoid special characters in Name, Author, and Details"); 
+        } else if (typeof data.year_created !== "number" || typeof data.commits !== "number") {
+            res.status(400).send("Please use numbers when providing commits and year created"); 
```

Comment:
create an express middleware to handle validation logic and use the middleware chaining.

Reply:
Made the changes by including a repovalidation middleware function and chained it to the create-repo route